v3/src/components/graph/components/graph.tsx

```diff
@@ -122,20 +122,46 @@ export const Graph = observer(function Graph({graphController, graphRef, pixiPoi
       {name: "Graph.handleAttributeConfigurationChange"}, graphModel)
   }, [graphController, graphModel])
 
-  useEffect(function handleDeleteAttribute() {
-    return dataset && addDisposer(dataset, onAnyAction(dataset, action => {
-      if (isRemoveAttributeAction(action)) {
-        const [attrId] = action.args
-        graphModel.dataConfiguration.rolesForAttribute(attrId).forEach(role => {
-          if (role === "yPlus") {
-            graphModel.dataConfiguration.removeYAttributeWithID(attrId)
-          } else {
-            graphModel.setAttributeID(role as GraphAttrRole, "", "")
+  useEffect(function handleAttributeCollectionMapChange() {
+
+    const constructAttrCollections = () => {
+      const graphAttrs = graphModel.dataConfiguration.attributes
+      const attrCollections: Record<string, string> = {}
+      graphAttrs.forEach(attrId => {
+        const collection = dataset?.getCollectionForAttribute(attrId)?.id
+        collection && (attrCollections[attrId] = collection)
```

Comment:
```suggestion
        const collectionId = dataset?.getCollectionForAttribute(attrId)?.id
        collectionId && (attrCollections[attrId] = collectionId)
```
for naming consistency.